cli/central-client/src/main/java/org/ballerinalang/central/client/Utils.java

```diff
@@ -266,9 +276,9 @@ static void writeBalaFile(Response balaDownloadResponse, Path balaPath, String f
                             logFormatter.formatLog("error occurred copying the bala file: " + e.getMessage()));
                 }
                 try {
-                    extractBala(balaPath, Optional.of(balaPath.getParent()).get());
+                    extractBala(balaPath, Optional.of(balaPath.getParent()).get(), trueDigest, fullPkgName);
                     Files.delete(balaPath);
-                } catch (IOException e) {
+                } catch (Exception e) {
```

Comment:
Can we use the union of the exact exceptions? eg : IOException | CentralClientException

Reply:
Fix in https://github.com/ballerina-platform/ballerina-lang/pull/41490/commits/8502a363b75e58a56c9163ad14036facc8bdd74e